src/components/common/Card/Card.jsx

```diff
@@ -2,25 +2,24 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { withRouter } from 'react-router-dom';
 import uuidv4 from 'uuid/v4';
+import _ from 'lodash';
 import { Icon } from 'semantic-ui-react';
 
 import CardContent from './CardContent';
 
 import '../../../_css/Card.css';
 
 export class Card extends React.Component {
-  handleView = (viewDetailsRoute, data) => {
-    const { history } = this.props;
-
-    if (!viewDetailsRoute) {
-      return null;
+  handleView = (viewDetailsRoute, data, event) => {
+    if (event.target.nodeName === 'DIV' && !_.isEmpty(viewDetailsRoute)) {
```

Comment:
It'd be better if `event.target.nodeName === 'DIV'` is assigned to a variable. Example, `hasClickedOnDepartmentDiv`

Reply:
I wonder why we are inspecting the `nodeName` though. Care to explain?

Reply:
Let's fix this so that we don't have to inspect `nodeName`

Reply:
It is necessary to inspect nodeName because the `Icon` is in the `div` and a click on the Icon usually calls the onClick function on the div.